src/main/java/com/brooklynotter/stucktogether/data/SphereTeam.java

```diff
@@ -0,0 +1,160 @@
+package com.brooklynotter.stucktogether.data;
+
+import com.brooklynotter.stucktogether.util.nbt.NBTHelper;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.StringTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.util.INBTSerializable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.util.List;
+import java.util.*;
+
+public class SphereTeam implements INBTSerializable<CompoundTag> {
+
+    protected @Nullable String teamName;
+    protected @Nullable Color sphereColor; // For future uses, maybe? :P
+    protected @NotNull UUID leader;
+    protected @NotNull Set<UUID> members;
+
+    public SphereTeam(CompoundTag tag) {
+        this.leader = new UUID(0, 0);
+        this.members = new HashSet<>();
+        deserializeNBT(tag);
+    }
+
+    public SphereTeam(@NotNull UUID leader) {
+        this.leader = leader;
+        this.members = new HashSet<>(Collections.singleton(leader));
+    }
+
+    @Nullable
+    public String getTeamName() {
+        return teamName;
+    }
+
+    @Nullable
+    public Color getSphereColor() {
+        return sphereColor;
+    }
+
+    @NotNull
+    public UUID getLeader() {
+        return leader;
+    }
+
+    @NotNull
+    public Set<UUID> getMembers() {
+        return Collections.unmodifiableSet(members);
+    }
+
+    public List<ServerPlayer> getOnlineMembers(MinecraftServer server) {
+        return members.stream()
+                .map(server.getPlayerList()::getPlayer)
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    public BlockPos getCenter(MinecraftServer server) {
+        float x = 0, y = 0, z = 0;
+        int onlineCount = 0;
```

Comment:
Well, `onlineCount` should actually be the length of `getOnlineMembers(server)`, so I probably should've extracted that to a variable,and used its length directly, instead of counting manually as I iterate. *Late night refactors be like* XD